crates/remora/src/executor/fake.rs

```diff
@@ -0,0 +1,196 @@
+use rand::thread_rng;
+use rand_distr::{Distribution, Normal};
+
+use sui_types::base_types::VersionNumber;
+use super::api::{ExecutableTransaction, ExecutionResults, Executor, StateStore, Transaction};
+use sui_single_node_benchmark::benchmark_context::BenchmarkContext;
+use sui_types::{
+    base_types::{ObjectID, ObjectRef, SequenceNumber}, committee::EpochId, digests::TransactionDigest, effects::TransactionEffects, error::SuiResult, object::{Object, Owner}, storage::{BackingPackageStore, ChildObjectResolver, ObjectStore, PackageObject, ParentSync}, transaction::InputObjectKind
+};
+
+
+#[derive(Clone)]
+pub struct FakeTransaction {
+    delay: u64, // delay in ms for that this transaction simulates
+}
+pub type FakeTransactionType = Transaction<FakeExecutor>;
+
+pub type FakeExecutionResults = ExecutionResults<FakeExecutor>;
+pub struct FakeStore;
+
+impl FakeExecutionResults {
+    // todo: is there a way to make this a constant instead of a fn?
+    pub fn dummy() -> Self {
+        FakeExecutionResults {
+            updates: TransactionEffects::default(),
+            new_state: BTreeMap::new(),
+        }
+    }
+}
+
+impl ExecutableTransaction for FakeTransaction {
+    fn digest(&self) -> &TransactionDigest {
+        // Return a reference to a static TransactionDigest
+        &TransactionDigest::ZERO
+    }
+
+    fn input_objects(&self) -> Vec<InputObjectKind> {
+        // Return an empty vector since the method won't be used
+        Vec::new()
+    }
+}
+
+impl ObjectStore for FakeStore {
+    fn get_object(
+        &self,
+        _object_id: &ObjectID,
+    ) -> Result<Option<Object>, sui_types::storage::error::Error> {
+        unreachable!()
+    }
+
+    fn get_object_by_key(
+        &self,
+        _object_id: &ObjectID,
+        _version: VersionNumber,
+    ) -> Result<Option<Object>, sui_types::storage::error::Error> {
+        unreachable!()
+    }
+}
+
+impl ParentSync for FakeStore {
+    fn get_latest_parent_entry_ref_deprecated(
+        &self,
+        _object_id: ObjectID,
+    ) -> SuiResult<Option<ObjectRef>> {
+        unreachable!()
+    }
+}
+
+impl ChildObjectResolver for FakeStore {
+    fn read_child_object(
+        &self,
+        parent: &ObjectID,
+        child: &ObjectID,
+        child_version_upper_bound: SequenceNumber,
+    ) -> SuiResult<Option<Object>> {
+        Ok(self.get_object(child).unwrap().and_then(|o| {
+            if o.version() <= child_version_upper_bound
+                && o.owner == Owner::ObjectOwner((*parent).into())
+            {
+                Some(o.clone())
+            } else {
+                None
+            }
+        }))
+    }
+
+    fn get_object_received_at_version(
+        &self,
+        _owner: &ObjectID,
+        _receiving_object_id: &ObjectID,
+        _receive_object_at_version: SequenceNumber,
+        _epoch_id: EpochId,
+    ) -> SuiResult<Option<Object>> {
+        unimplemented!()
+    }
+}
+
+impl BackingPackageStore for FakeStore {
```

Comment:
I suspect that this project will not support publishing smart contract (we only load them upon booting)